File: Python/Lab_2/exhibits/Sword.py
from exhibits.Exhibit import *
from enums.ExhibitName import *
from enums.ExhibitAge import *


class Sword(Exhibit):
    exhibit_age = ExhibitAge.FIVE_BC
    exhibit_name = ExhibitName.SWORD

    def __init__(self, sword_length, name, age, kind_of_art, place_of_manufacture, condition, weight):
        self.sword_length = sword_length
        Exhibit.__init__(self, name, age, kind_of_art, place_of_manufacture, condition, weight)

    def __str__(self):
        return "Name : " + str(self.name) + " age: " + str(self.age) + " the length of sword is " + str(
            self.sword_length)

File: Lab_5/src/main/java/ua/Lviv/iot/Sword.java
package ua.Lviv.iot;

public class Sword extends Exhibit {
    private double swordLength;

    public Sword() {

    }

    public Sword(ExhibitName name, ExhibitAge age, String kindOfArt, String placeOfManufacture, String condition, double weight, double swordLength) {
        super(name, age, kindOfArt, placeOfManufacture, condition, weight);
        this.swordLength = swordLength;
    }

    public double getSwordLength() {
        return swordLength;
    }

    public void setSwordLength(double swordLength) {
        this.swordLength = swordLength;
    }

    public String getHeaders() {
        return super.getHeaders() + "," + "swordLength";
    }

    public String toCVS() {
        return super.toCVS() + "," + getSwordLength();
    }

    @Override
    public String toString() {
        return super.toString() + ", sword length: " + this.swordLength + "\n";
    }

}